modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java

```diff
@@ -437,6 +437,19 @@ public void addQueue(String storageQueueName) throws AndesException {
      * {@inheritDoc}
      */
     @Override
+    public Map<String, Integer> getMessageCountForAllQueues(List<String> queueNames) throws AndesException {
+        try {
+            return wrappedInstance.getMessageCountForAllQueues(queueNames);
+        } catch (AndesStoreUnavailableException exception) {
+            notifyFailures(exception);
```

Comment:
Yeah we can. But if we create a notifyFailuresAndThrow with the return type as void, we'll again need to explicitly return an object after calling it. For example, the getMessageCountForAllQueues method will look as follows:

public Map<String, Integer> getMessageCountForAllQueues(List<String> queueNames) throws AndesException {
        try {
            return wrappedInstance.getMessageCountForAllQueues(queueNames);
        } catch (AndesStoreUnavailableException exception) {
            notifyFailuresAndThrow(exception);
            return null;
        }
    }

This adds an unnecessary overhead.